test/recipes/30-test_evp_data/evpkdf.txt

```diff
@@ -252,4 +252,63 @@ Ctrl.IKM = hexkey:0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c0c
 Ctrl.salt = salt:
 Output = 2c91117204d745f3500d636a62f64f0ab3bae548aa53d423b0d1f27ebba6f5e5673a081d70cce7acfc48
 
-
+Title = id-scrypt tests (from draft-josefsson-id-scrypt-kdf-03 and others)
+
+# Empty password/salt currently unsupported by PKEY KDF scrypt
+#KDF = scrypt
+#Ctrl.pass = pass:""
+#Ctrl.salt = salt:""
+#Ctrl.N = N:16
+#Ctrl.r = r:1
+#Ctrl.p = p:1
+#Output = 77d6576238657b203b19ca42c18a0497f16b4844e3074ae8dfdffa3fede21442fcd0069ded0948f8326a753a0fc81f17e8d3e0fb2e0d3628cf35e20c38d18906
+
```

Comment:
I think it's important that we find a way to support this. The reason is that in future I'd like to make EVP_PKEY_METHOD the preferred interface for scrypt (and other PBE algorithms) so it must be at least as flexible as the underlying function. I can think of several ways to do that, one is to allocate the fields when set even if they're zero length so the != NULL check passes.